src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/ExceptionHandling.cs

```diff
@@ -631,11 +552,7 @@ public static void RhThrowEx(object exceptionObj, ref ExInfo exInfo)
             InternalCalls.RhpValidateExInfoStack();
 #endif
             // Transform attempted throws of null to a throw of NullReferenceException.
-            if (exceptionObj == null)
-            {
-                IntPtr faultingCodeAddress = exInfo._pExContext->IP;
-                exceptionObj = GetClasslibException(ExceptionIDs.NullReference, faultingCodeAddress);
-            }
+            exceptionObj ??= new NullReferenceException();
```

Comment:
I think this should stay as is. It is same case as `GetClasslibException` in `RhThrowHwEx`